integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,158 @@
-// go:build linux && integration
+//go:build linux && integration
 // +build linux,integration
 
 package performancetest
 
-import(
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"sync"
 	"testing"
 	"time"
-	"log"
-	"context"
+
 	"github.com/aws/amazon-cloudwatch-agent/integration/test"
 )
 
 const (
-	configPath = "resources/config.json"
 	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
-	agentRuntimeMinutes = 5 //20min desired but 5mins for testing purposes 
+	agentRuntimeMinutes = 5 //20 mins desired but 5 mins for testing purposes
 	DynamoDBDataBase = "CWAPerformanceMetrics"
-
 )
 
 func TestPerformance(t *testing.T) {
+	configFiles := []string {
+		"./resources/config10Logs.json",
+		"./resources/config100Logs.json",
+		"./resources/config1000Logs.json"
+	}
+
+	tpsVals := []int {
+		10,
+		100,
+		1000
+	}
+	
 	agentContext := context.TODO()
 	instanceId := test.GetInstanceId()
 	log.Printf("Instance ID used for performance metrics : %s\n", instanceId)
 
-	test.CopyFile(configPath, configOutputPath)
+	//data base
+	dynamoDB := InitializeTransmitterAPI(DynamoDBDataBase) //add cwa version here
+	if dynamoDB == nil {
+		t.Fatalf("Error: generating dynamo table")
+	}
+
+	//run tests
+	for _, configPath := range configFiles {
+		for _, tps := range tpsVals {
+			t.Run(fmt.Sprintf("config file location: %s", configPath), func(t *testing.T) {
+				test.CopyFile(configPath, configOutputPath)
+	
+				test.StartAgent(configOutputPath, true)
+	
+				agentRunDuration := agentRuntimeMinutes * time.Minute
```

Comment:
Would it better to initialize the 
```
agentRuntimeMinutes := 5 * time.Minute
```
instead of initialize it here? To avoid creating shallow variables and recreate the variable all the time (pointers and memory perhaps)

Reply:
I think leaving it as a constant at the top is better because it makes it easier to change and propagate through the code, especially because the 5 is temporary right now

Reply:
It's the same though. `const agentRuntimeMinutes = 20 * time.Minute` at the top is better than initialize inside during the run so I'm not convinced it would be better to change the variable inside the run.